server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/transformer/ContinuationLineTransformation.java

```diff
@@ -85,12 +86,85 @@ private SyntaxError checkContinuationLine(String uri, int lineNumber, CobolLine
+          checkCompilerDirectiveContinued(cobolLine, uri, lineNumber);
+      if (Objects.nonNull(compilerDirectiveContinuedError)) {
+        return compilerDirectiveContinuedError;
+      }
+
       // invoke method for noContentInAreaA
       return checkContentAreaAWithContinuationLine(cobolLine, uri, lineNumber);
     }
     return null;
   }
 
+  private SyntaxError checkCompilerDirectiveContinued(
+      CobolLine cobolLine, String uri, int lineNumber) {
+    if (isCompilerDirectiveStatement(cobolLine)) {
+      return SyntaxError.syntaxError()
+          .severity(ERROR)
+          .locality(
+              Locality.builder()
+                  .uri(uri)
+                  .range(
+                      new Range(
+                          new Position(lineNumber, ProcessingConstants.INDICATOR_AREA),
+                          new Position(lineNumber, cobolLine.toString().length())))
+                  .recognizer(ContinuationLineTransformation.class)
+                  .build())
+          .suggestion(
+              messageService.getMessage(
+                  "ContinuationLineTransformation.compilerDirectiveContinued"))
+          .build();
+    }
+    return null;
+  }
+
+  private boolean isCompilerDirectiveStatement(CobolLine line) {
+    if (Objects.isNull(line.getPredecessor())) return false;
+    String predecessorContentArea = line.getPredecessor().getContentArea();
+    return Stream.of(CompilerDirectives.values())
+            .map(CompilerDirectives::getDirective)
+            .anyMatch(predecessorContentArea.split("\\s+")[0].toUpperCase()::equals)
+        || isPseudoDelimiterContinued(line, predecessorContentArea);
+  }
+
+  private boolean isPseudoDelimiterContinued(CobolLine line, String predecessorContentArea) {
+    if (isQuotesBalanced(predecessorContentArea)) {
+      String inspectionText =
+          StringUtils.normalizeSpace(line.getContentArea()).charAt(0)
+              + getLastValidChar(predecessorContentArea);
+      return inspectionText.equals(PSEUDO_TEXT_DELIMITER);
+    }
+    // If quotes are unbalanced, it implies its a string continuation.
+    return false;
+  }
+
+  private String getLastValidChar(String predecessorContentArea) {
+    String normalizedContent = StringUtils.normalizeSpace(predecessorContentArea);
+    String lastValidChar = normalizedContent.substring(normalizedContent.length() - 1);
+
+    // consider delimiters in a continue line.
+    if (StringUtils.endsWithAny(normalizedContent, ",", ";"))
+      lastValidChar = predecessorContentArea.substring(normalizedContent.length() - 2);
+    return lastValidChar;
+  }
+
+  private boolean isQuotesBalanced(String predecessorContentArea) {
+    int quotesCount = StringUtils.countMatches(predecessorContentArea, "\"");
+    int singleQuotesCount = StringUtils.countMatches(predecessorContentArea, "'");
+
+    return isEven(quotesCount)
+        || isEven(singleQuotesCount)
+        || (quotesCount + singleQuotesCount == 0);
```

Comment:
Why `||`? Do you need all types of quotes to be balanced or just one?

BTW, you can replace `isEven(quotesCount)` by `quotesCount % 2 == 0`.

Reply:
👍  isEven was stupid 😄 
I am re-thinking on this. Need to check the below statements
```
"Hello, I don't know"

' "Hello" , I don't know ' 
```

Reply:
Done. Remove that logic completely.
ref: https://www.ibm.com/support/knowledgecenter/SS6SG3_6.1.0/lr/ref/rlfmtcon.html
Topic: Continuation of alphanumeric and national literals